public/components/trace_analytics/requests/request_handler.ts

```diff
@@ -35,6 +36,50 @@ export async function handleDslRequest(
   const query = {
     dataSourceMDSId,
   };
+
+  const handleError = (error: any) => {
```

Comment:
this function might need unit tests to make it easier to understand the examples of `error` variable, and to ensure there's no regression

Reply:
Extraced handleError to a helper_functions file and included tested for it
public/components/trace_analytics/requests/__tests__/helper_funtions.test.tsx

```
 PASS  public/components/trace_analytics/requests/__tests__/helper_funtions.test.tsx (5.281 s)
  handleError in handleDslRequest
    ✓ displays danger toast for too_many_buckets_exception (1 ms)
    ✓ logs error for non-bucket exceptions
    ✓ parses nested message JSON string (1 ms)